src/context/AppContext.context.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useEffect, useState } from 'react';
+
+import type { AppContextProps, ColorScheme } from '@/interfaces/AppContextProps.interface.tsx';
+
+export const AppContext = createContext<AppContextProps>({
+    theme: 'dark',
+    setTheme: (mode: ColorScheme) => {},
+});
+
+interface AppContextProviderProps {
+    children: React.ReactNode;
+}
+
+export const AppContextProvider: React.FC<AppContextProviderProps> = ({ children }) => {
+    const browserThemeMode: ColorScheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+
+    const [theme, setTheme] = useState<ColorScheme>(() => {
+        const savedScheme = localStorage.getItem('theme');
+        const validColorSchemes: ColorScheme[] = ['dark', 'light'];
+
+        return savedScheme && validColorSchemes.includes(savedScheme as ColorScheme) ? (savedScheme as ColorScheme) : browserThemeMode;
+    });
+
+    useEffect(() => {
+        const removeThemeClasses = ['dark', 'light'];
+
+        if (browserThemeMode === theme) {
+            document.documentElement.classList.remove(...removeThemeClasses);
```

Comment:
it's bad approach to work with DOM like this in React, you write some extra logic. If you want work with Context use example:

https://codesandbox.io/p/sandbox/theme-toggle-h54n8?file=%2Fsrc%2Findex.js&from-embed=